sshd-auth.c

```diff
@@ -588,7 +588,7 @@ main(int ac, char **av)
 	if (!rexeced_flag)
 		fatal("sshd-auth should not be executed directly");
 
-#ifdef WITH_OPENSSL
+#if defined(WITH_OPENSSL) && defined(HAVE_OPENSSL_ADD_ALL_ALGORITHMS)
 	OpenSSL_add_all_algorithms();
```

Comment:
I think this could be dropped entirely both in portable (where the compat layer if needed via seed_rng() -> ssh_libcrypto_init() -> ssh_libcrypto_init()) and upstream (where I don't think it's needed any more).  I'll check with them.

I *think* the call to OpenSSL_add_all_algorithms snuck back in from OpenBSD via the sshd process split, since it was previously delegated to ssh_libcrypto_init in Portable via commit 42c5ec4b97b6a1bae70f323952d0646af16ce710

Reply:
thank you for the review! I will trace/verify all of the above, and send a new patch.

Reply:
Having checked with the LibreSSL folks, the state seems to be:
 - OPENSSL_init_crypto is in the oldest OpenSSL (1.1.1) and LibreSSL (3.1) that we support, so we can drop the configure checks.
 -  OpenSSL_add_all_algorithms is a no-op in current OpenBSD & LibreSSL versions.

There is no need to send an updated patch, the changes will mostly be in upstream OpenBSD, with a resync and cleanup in portable.  You are welcome to verify the things said here, though.